### PR TITLE
Integrate `@formatjs/intl` as a replacement for `t()`

### PR DESCRIPTION
**Pull request by Etheryte**
## What does this PR change?

Our current translation wrapper `t()` is a simple hand-rolled solution that works for basic translation, but notably does not support translations with DOM nodes mixed in etc. This PR integrates [`@formatjs/intl`](https://formatjs.io/) as a near drop-in replacement to add support for the missing functionality and simplify many of our existing translations.  

`@formatjs/intl`, the library internally used by `react-intl`, was picked over a vanilla `react-intl` setup to accommodate for our SPA architecture around `senna` which makes it difficult to use global wrappers such as intl providers. If we ever drop `senna`, we can easily swap out `@formatjs/intl` for `react-intl`.

### Notable changes:

No more numeric replacements, all replacements **must** now be named.  

Before:

```tsx
t("Items {0} - {1} of {2}", from, to, total);
```

After:  

```tsx
t("Items {from} - {to} of {total}", { from, to, total });
```

Support for mixing DOM and React components into translations has been added, meaning no more chopped up translations.

Before:
```tsx
<p>
  {t("Successfully bootstrapped host! Your system should appear in ")}
  <a className="js-spa" href="/rhn/manager/systems/list/all">
    {t("systems")}
  </a>
  {t(" shortly. If it is a transactional system, please reboot it to finish registration.")}
</p>
```

After:

```tsx
<p>
  {t(
    "Successfully bootstrapped host! Your system should appear in <link>systems</link> shortly. If it is a transactional system, please reboot it to finish registration.",
    {
      link: (str) => (
        <a className="js-spa" href="/rhn/manager/systems/list/all">
          {str}
        </a>
      ),
    }
  )}
</p>
```

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- Unit tests were added

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/20449
Tracks https://github.com/SUSE/spacewalk/issues/21208

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: web/html/src/core/spa/spa-renderer.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ window.pageRenderers.spa.reactRenderers = window.pageRenderers.spa.reactRenderer
 window.pageRenderers.spa.previousReactRenderers = window.pageRenderers.spa.previousReactRenderers || [];
 
 function addReactApp(appName: string) {
-  getTranslationData();
+  // getTranslationData();
```
Comment (parlt91) [inline web/html/src/core/spa/spa-renderer.ts:20]: Can be removed

File: web/html/src/core/spa/spa-renderer.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ function renderGlobalReact(element: JSX.Element, container: Element | null | und
     throw new Error("The DOM element is not present.");
   }
 
-  getTranslationData();
+  // getTranslationData();
```
Comment (parlt91) [inline web/html/src/core/spa/spa-renderer.ts:33]: Same here

**Review by parlt91 — APPROVED** [review]
This is a really valuable addition. TYVM
See two really small remarks below

File: web/html/src/core/spa/spa-renderer.ts (hunk 1 above)
Comment (Etheryte) [reply web/html/src/core/spa/spa-renderer.ts]: Thanks for the heads up, I missed those.

**Pull request merged**